File: backend/app/schemas/destination.py
from datetime import datetime
from pydantic import computed_field
from pydantic import BaseModel, ConfigDict


class DestinationCreate(BaseModel):
    name: str
    description: str
    division: str
    district: str


class DestinationUpdate(BaseModel):
    name: str | None = None
    description: str | None = None
    division: str | None = None
    district: str | None = None


class DestinationResponse(BaseModel):
    id: int
    name: str
    description: str
    division: str
    district: str
    image: str
    created_at: datetime
    updated_at: datetime

    @computed_field
    @property
    def image_url(self) -> str:
        return f"http://127.0.0.1:8000/uploads/destination_images/{self.image}"

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/destination.ts
export interface Destination {
  id: number;
  name: string;
  description: string;
  division: string;
  district: string;
  image: string;
  created_at: string;
  updated_at: string;
}

export interface DestinationCreate {
  name: string;
  description: string;
  division: string;
  district: string;
  image: File;
}

export interface DestinationUpdate {
  name?: string;
  description?: string;
  division?: string;
  district?: string;
  image?: File;
}